core/src/test/java/org/apache/iceberg/TestDataTableScan.java

```diff
@@ -130,4 +132,63 @@ public void testTableScanWithPlanExecutor() {
     Assert.assertEquals(2, Iterables.size(scan.planFiles()));
     Assert.assertTrue("Thread should be created in provided pool", planThreadsIndex.get() > 0);
   }
+
+  @Test
+  public void testReaddingPartitionField() throws Exception {
+    Assume.assumeTrue(formatVersion == 2);
```

Comment:
What happens with v1?

Reply:
The change itself only affects V2. Since in V1 the reused partition field names always get appended by the ID, the fix is a no-op there. Another reason for this assume() call is that expected file structure differs greatly in V1 due to the void transformations that deleted partition fields get converted to, the test code would be quite bloated :/

Reply:
Right. Btw I spotted an unnecessary if statement in your example: https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java#L385
I guess all these could be cleaned up in another PR.